src/main/java/com/r307/arbitrader/service/ExchangeService.java

```diff
@@ -40,10 +41,14 @@
     private final ExchangeFeeCache feeCache;
     private final TickerStrategyProvider tickerStrategyProvider;
 
+    /** The current bankroll. Holds the sum of all account's balance. Useful to have a broad idea how our bankroll is evolving */
+    private final AtomicReference<BigDecimal> combinedBalance;
+
     @Inject
     public ExchangeService(ExchangeFeeCache feeCache, TickerStrategyProvider tickerStrategyProvider) {
         this.feeCache = feeCache;
         this.tickerStrategyProvider = tickerStrategyProvider;
+        this.combinedBalance = new AtomicReference<>(BigDecimal.ZERO);
```

Comment:
Storing this as a single `BigDecimal` is going to be difficult to maintain over time. I'd suggest storing a `BigDecimal` on each `Exchange` with that exchange's last known balance, then when you need the combined value go through and add them together.
```
List<BigDecimal> balances = allExchanges.stream().map(exchange -> exchange.getLastKnownBalance()).collect(Collectors.toList());
BigDecimal combinedBalance = BigDecimal.ZERO;

for (BigDecimal balance : balances) {
  combinedBalance = combinedBalance.add(balance);
}
```
(I'm sure there's a fancy way to add the BigDecimals together as part of the stream operation but it's not coming to me.)

This way each time we query for a balance it simply replaces the old value instead of applying a change to the combined value. It should be less likely to become inaccurate due to rounding errors and stuff like that over time.

And actually it occurs to me as I'm writing this that we already have an `ExchangeBalanceCache` that would be a perfect home for the "add them all together" method because it should already have all the last known balances for all the exchanges in it.